File: src/main/java/moving/features/ChampionsLeague.java
package moving.features;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;
import java.util.stream.Collectors;

@AllArgsConstructor
@Getter
public class ChampionsLeague {
    private final List<Team> teams;

    public String nextStageTable() {
        return teams.stream()
                .map(team -> team.getName() + " - " + team.getCountry())
                .collect(Collectors.joining("\n"));
    }
}

File: src/scala/src/main/scala/moving/features/ChampionsLeague.scala
package moving.features

import scala.util.Properties.lineSeparator

final case class ChampionsLeague(teams: List[Team]) {
  lazy val nextStageTable: String =
    teams
      .map(team => s"${team.name} - ${team.country}")
      .mkString(lineSeparator)
}
